Linkedit error on OS X

When building on OS X I get:

```
==> Processing catkin package: 'moveit_setup_assistant'
==> Building with env: '/Users/william/moveit_ws/install_isolated/env.sh'
Makefile exists, skipping explicit cmake invocation...
==> make cmake_check_build_system in '/Users/william/moveit_ws/build_isolated/moveit_setup_assistant'
==> make -j1 in '/Users/william/moveit_ws/build_isolated/moveit_setup_assistant'
Linking CXX shared library libmoveit_setup_assistant_tools.dylib
[  6%] Built target moveit_setup_assistant_tools
Linking CXX shared library libmoveit_setup_assistant_widgets.dylib
Undefined symbols for architecture x86_64:
  "SetupScreenWidget::highlightLink(std::string const&)", referenced from:
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedLink(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedJoints(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::DefaultCollisionsWidget::previewClicked(int, int) in default_collisions_widget.cpp.o
      moveit_setup_assistant::PassiveJointsWidget::previewSelectedJoints(std::vector<std::string, std::allocator<std::string> >) in passive_joints_widget.cpp.o
  "SetupScreenWidget::highlightGroup(std::string const&)", referenced from:
      moveit_setup_assistant::PlanningGroupsWidget::previewSelected() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedSubgroup(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::loadJointSliders(QString const&) in robot_poses_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::previewClicked(int, int) in robot_poses_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::showPose(srdf::Model::GroupState*) in robot_poses_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::playPoses() in robot_poses_widget.cpp.o
      moveit_setup_assistant::EndEffectorsWidget::previewClicked(int, int) in end_effectors_widget.cpp.o
      ...
  "SetupScreenWidget::unhighlightAll()", referenced from:
      moveit_setup_assistant::PlanningGroupsWidget::previewSelected() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedLink(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedJoints(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::previewSelectedSubgroup(std::vector<std::string, std::allocator<std::string> >) in planning_groups_widget.cpp.o
      moveit_setup_assistant::DefaultCollisionsWidget::previewClicked(int, int) in default_collisions_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::loadJointSliders(QString const&) in robot_poses_widget.cpp.o
      moveit_setup_assistant::RobotPosesWidget::previewClicked(int, int) in robot_poses_widget.cpp.o
      ...
  "SetupScreenWidget::staticMetaObject", referenced from:
      moveit_setup_assistant::SetupAssistantWidget::navigationClicked(QModelIndex const&) in setup_assistant_widget.cpp.o
      moveit_setup_assistant::SetupAssistantWidget::moveToScreen(int) in setup_assistant_widget.cpp.o
      moveit_setup_assistant::SetupAssistantWidget::progressPastStartScreen() in setup_assistant_widget.cpp.o
  "SetupScreenWidget::isModal(bool)", referenced from:
      moveit_setup_assistant::PlanningGroupsWidget::PlanningGroupsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::showMainScreen() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::editSelected() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::changeScreen(int) in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::deleteGroup() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::addGroup() in planning_groups_widget.cpp.o
      moveit_setup_assistant::PlanningGroupsWidget::saveJointsScreen() in planning_groups_widget.cpp.o
      ...
  "moveit_setup_assistant::SliderWidget::jointValueChanged(std::string const&, double)", referenced from:
      moveit_setup_assistant::SliderWidget::changeJointValue(int) in robot_poses_widget.cpp.o
      moveit_setup_assistant::SliderWidget::changeJointSlider() in robot_poses_widget.cpp.o
  "moveit_setup_assistant::DoubleListWidget::previewSelected(std::vector<std::string, std::allocator<std::string> >)", referenced from:
      moveit_setup_assistant::DoubleListWidget::previewSelected(QList<QTableWidgetItem*> const&) in double_list_widget.cpp.o
  "moveit_setup_assistant::DoubleListWidget::selectionUpdated()", referenced from:
      moveit_setup_assistant::DoubleListWidget::selectDataButtonClicked() in double_list_widget.cpp.o
      moveit_setup_assistant::DoubleListWidget::deselectDataButtonClicked() in double_list_widget.cpp.o
  "moveit_setup_assistant::StartScreenWidget::readyToProgress()", referenced from:
      moveit_setup_assistant::StartScreenWidget::loadNewFiles() in start_screen_widget.cpp.o
      moveit_setup_assistant::StartScreenWidget::loadExistingFiles() in start_screen_widget.cpp.o
  "moveit_setup_assistant::StartScreenWidget::loadRviz()", referenced from:
      moveit_setup_assistant::StartScreenWidget::loadNewFiles() in start_screen_widget.cpp.o
      moveit_setup_assistant::StartScreenWidget::loadExistingFiles() in start_screen_widget.cpp.o
  "moveit_setup_assistant::VirtualJointsWidget::referenceFrameChanged()", referenced from:
      moveit_setup_assistant::VirtualJointsWidget::doneEditing() in virtual_joints_widget.cpp.o
  "moveit_setup_assistant::KinematicChainWidget::highlightLink(std::string const&)", referenced from:
      moveit_setup_assistant::KinematicChainWidget::itemSelected() in kinematic_chain_widget.cpp.o
  "moveit_setup_assistant::KinematicChainWidget::unhighlightAll()", referenced from:
      moveit_setup_assistant::KinematicChainWidget::itemSelected() in kinematic_chain_widget.cpp.o
  "boost::program_options::abstract_variables_map::operator[](std::string const&) const", referenced from:
      moveit_setup_assistant::SetupAssistantWidget::SetupAssistantWidget(QWidget*, boost::program_options::variables_map) in setup_assistant_widget.cpp.o
  "vtable for moveit_setup_assistant::NavDelegate", referenced from:
      moveit_setup_assistant::NavigationWidget::NavigationWidget(QWidget*) in navigation_widget.cpp.o
      moveit_setup_assistant::NavDelegate::NavDelegate(QObject*) in navigation_widget.cpp.o
      moveit_setup_assistant::NavDelegate::NavDelegate(QObject*) in navigation_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::HeaderWidget", referenced from:
      moveit_setup_assistant::HeaderWidget::HeaderWidget(std::string const&, std::string const&, QWidget*) in header_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::SliderWidget", referenced from:
      moveit_setup_assistant::SliderWidget::SliderWidget(QWidget*, robot_model::JointModel const*, double) in robot_poses_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::LoadPathWidget", referenced from:
      moveit_setup_assistant::LoadPathWidget::LoadPathWidget(std::string const&, std::string const&, bool, bool, QWidget*) in header_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::GroupEditWidget", referenced from:
      moveit_setup_assistant::GroupEditWidget::GroupEditWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in group_edit_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::DoubleListWidget", referenced from:
      moveit_setup_assistant::DoubleListWidget::DoubleListWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>, QString, QString, bool) in double_list_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::NavigationWidget", referenced from:
      moveit_setup_assistant::NavigationWidget::NavigationWidget(QWidget*) in navigation_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::RobotPosesWidget", referenced from:
      moveit_setup_assistant::RobotPosesWidget::RobotPosesWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in robot_poses_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::SelectModeWidget", referenced from:
      moveit_setup_assistant::SelectModeWidget::SelectModeWidget(QWidget*) in start_screen_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::StartScreenWidget", referenced from:
      moveit_setup_assistant::StartScreenWidget::StartScreenWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in start_screen_widget.cpp.o
      moveit_setup_assistant::StartScreenWidget::~StartScreenWidget() in start_screen_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::EndEffectorsWidget", referenced from:
      moveit_setup_assistant::EndEffectorsWidget::EndEffectorsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in end_effectors_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::PassiveJointsWidget", referenced from:
      moveit_setup_assistant::PassiveJointsWidget::PassiveJointsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in passive_joints_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::VirtualJointsWidget", referenced from:
      moveit_setup_assistant::VirtualJointsWidget::VirtualJointsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in virtual_joints_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::KinematicChainWidget", referenced from:
      moveit_setup_assistant::KinematicChainWidget::KinematicChainWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in kinematic_chain_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::PlanningGroupsWidget", referenced from:
      moveit_setup_assistant::PlanningGroupsWidget::PlanningGroupsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in planning_groups_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::SetupAssistantWidget", referenced from:
      moveit_setup_assistant::SetupAssistantWidget::SetupAssistantWidget(QWidget*, boost::program_options::variables_map) in setup_assistant_widget.cpp.o
      moveit_setup_assistant::SetupAssistantWidget::~SetupAssistantWidget() in setup_assistant_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::DefaultCollisionsWidget", referenced from:
      moveit_setup_assistant::DefaultCollisionsWidget::DefaultCollisionsWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in default_collisions_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for moveit_setup_assistant::ConfigurationFilesWidget", referenced from:
      moveit_setup_assistant::ConfigurationFilesWidget::ConfigurationFilesWidget(QWidget*, boost::shared_ptr<moveit_setup_assistant::MoveItConfigData>) in configuration_files_widget.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libmoveit_setup_assistant_widgets.dylib] Error 1
make[1]: *** [CMakeFiles/moveit_setup_assistant_widgets.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,6 +78,7 @@ add_library(${PROJECT_NAME}_widgets
   src/widgets/header_widget.cpp
   src/widgets/setup_assistant_widget.cpp
   src/widgets/setup_screen_widget.cpp
+  ${moc_sources}
 )
 target_link_libraries(${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools ${QT_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 
@@ -90,16 +91,16 @@ catkin_package(
   INCLUDE_DIRS
     include
   LIBRARIES
-    ${PROJECT_NAME}_tools 
+    ${PROJECT_NAME}_tools
   CATKIN_DEPENDS
     moveit_ros_planning
     moveit_ros_visualization
     )
 
 # Link libraries
-add_executable(${PROJECT_NAME} ${sources} ${headers} ${moc_sources})
-target_link_libraries(${PROJECT_NAME} 
-  ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools 
+add_executable(${PROJECT_NAME} ${sources} ${headers})
+target_link_libraries(${PROJECT_NAME}
+  ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools
   ${QT_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES} log4cxx)
 
 install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools

--- a/src/widgets/configuration_files_widget.h
+++ b/src/widgets/configuration_files_widget.h
@@ -44,7 +44,11 @@
 #include <QLabel>
 #include <QListWidget>
 #include <QList>
+
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "header_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
 

--- a/src/widgets/default_collisions_widget.h
+++ b/src/widgets/default_collisions_widget.h
@@ -47,9 +47,13 @@
 #include <QProgressBar>
 #include <QCheckBox>
 #include <QSpinBox>
+
+#ifndef Q_MOC_RUN
 #include <boost/thread.hpp>
 #include <moveit/setup_assistant/tools/compute_default_collisions.h>
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "header_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
 

--- a/src/widgets/double_list_widget.h
+++ b/src/widgets/double_list_widget.h
@@ -40,7 +40,10 @@
 #include <QWidget>
 #include <QLabel>
 #include <QTableWidget>
+
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
 
 namespace moveit_setup_assistant
 {

--- a/src/widgets/end_effectors_widget.h
+++ b/src/widgets/end_effectors_widget.h
@@ -49,8 +49,12 @@
 #include <QStackedLayout>
 #include <QString>
 #include <QComboBox>
+
 // SA
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "header_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
 

--- a/src/widgets/group_edit_widget.h
+++ b/src/widgets/group_edit_widget.h
@@ -42,7 +42,10 @@
 #include <QLineEdit>
 #include <QComboBox>
 #include <QPushButton>
+
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
 
 namespace moveit_setup_assistant
 {

--- a/src/widgets/kinematic_chain_widget.h
+++ b/src/widgets/kinematic_chain_widget.h
@@ -40,8 +40,11 @@
 #include <QWidget>
 #include <QLabel>
 #include <QTreeWidget>
+
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
 #include <ros/ros.h>
+#endif
 
 namespace moveit_setup_assistant
 {

--- a/src/widgets/passive_joints_widget.h
+++ b/src/widgets/passive_joints_widget.h
@@ -47,8 +47,12 @@
 #include <QStackedLayout>
 #include <QString>
 #include <QComboBox>
+
 // SA
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "header_widget.h"
 #include "double_list_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant

--- a/src/widgets/planning_groups_widget.h
+++ b/src/widgets/planning_groups_widget.h
@@ -42,8 +42,12 @@
 #include <QTreeWidget>
 #include <QSplitter>
 #include <QStackedLayout>
+
 // Setup Asst
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "double_list_widget.h" // for joints, links and subgroups pages
 #include "kinematic_chain_widget.h" // for kinematic chain page
 #include "group_edit_widget.h" // for group rename page

--- a/src/widgets/robot_poses_widget.h
+++ b/src/widgets/robot_poses_widget.h
@@ -49,12 +49,16 @@
 #include <QStackedLayout>
 #include <QString>
 #include <QComboBox>
+
 // SA
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
 #include <moveit/planning_scene/planning_scene.h> // for collision stuff
+#include <ros/ros.h>
+#endif
+
 #include "header_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
-#include <ros/ros.h>
 
 namespace moveit_setup_assistant
 {

--- a/src/widgets/setup_assistant_widget.h
+++ b/src/widgets/setup_assistant_widget.h
@@ -59,11 +59,15 @@
 #include "virtual_joints_widget.h"
 #include "passive_joints_widget.h"
 #include "configuration_files_widget.h"
+
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+
 // Other
 #include <ros/ros.h>
 #include <boost/program_options.hpp> // for parsing input arguments
 #include <boost/thread/mutex.hpp>
+#endif
 
 // Forward declarations
 namespace rviz

--- a/src/widgets/start_screen_widget.h
+++ b/src/widgets/start_screen_widget.h
@@ -42,9 +42,13 @@
 #include <QPushButton>
 #include <QLabel>
 #include <QProgressBar>
+
+#ifndef Q_MOC_RUN
 #include <urdf/model.h> // for testing a valid urdf is loaded
 #include <srdfdom/model.h> // for testing a valid srdf is loaded
 #include <moveit/setup_assistant/tools/moveit_config_data.h> // common datastructure class
+#endif
+
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
 
 

--- a/src/widgets/virtual_joints_widget.h
+++ b/src/widgets/virtual_joints_widget.h
@@ -47,8 +47,12 @@
 #include <QStackedLayout>
 #include <QString>
 #include <QComboBox>
+
 // SA
+#ifndef Q_MOC_RUN
 #include <moveit/setup_assistant/tools/moveit_config_data.h>
+#endif
+
 #include "header_widget.h"
 #include "setup_screen_widget.h" // a base class for screens in the setup assistant
 

